File: desktop/LostieLauncher/Utils/ShutdownWarningPolicy.cs
namespace LostieLauncher.Utils;

public enum ShutdownWarning
{
    None,

    Download,

    Game,

    Both
}

public static class ShutdownWarningPolicy
{
    public static ShutdownWarning Decide(bool isDownloading, bool isGameRunning) => (isDownloading, isGameRunning) switch
    {
        (true, true) => ShutdownWarning.Both,
        (true, false) => ShutdownWarning.Download,
        (false, true) => ShutdownWarning.Game,
        _ => ShutdownWarning.None
    };
}

File: android/app/src/main/kotlin/dev/jagoba/lostielauncher/util/policy/ShutdownWarningPolicy.kt
package dev.jagoba.lostielauncher.util.policy

/** What the user has to be warned about before the launcher closes. */
enum class ShutdownWarning {
    None,
    Download,
    Game,
    Both,
}

/**
 * Decides which exit warning applies, ported from the desktop's
 * `Utils/ShutdownWarningPolicy.cs`.
 *
 * Member order matches the desktop's enum, as the layer rules require.
 */
object ShutdownWarningPolicy {
    fun decide(isDownloading: Boolean, isGameRunning: Boolean): ShutdownWarning = when {
        isDownloading && isGameRunning -> ShutdownWarning.Both
        isDownloading -> ShutdownWarning.Download
        isGameRunning -> ShutdownWarning.Game
        else -> ShutdownWarning.None
    }
}
